app/services/weather_service.py

```
@@ -28,216 +35,285 @@ def _get_cache(cls) -> ScopedCache:
+        if not (-90 <= lat <= 90):
+            raise ValueError(f"Invalid latitude: {lat}")
+        if not (-180 <= lon <= 180):
+            raise ValueError(f"Invalid longitude: {lon}")
+
+    @classmethod
+    def _handle_http_error(cls, e: httpx.HTTPError, lat: float, lon: float) -> None:
+        """Centralized error logging."""
+        if isinstance(e, httpx.HTTPStatusError):
+            if e.response.status_code == 401:
+                log_error(f"OpenWeather 401 Unauthorized for {lat},{lon}. Check API Key.")
+                raise ValueError("Invalid OpenWeather API Key.") from e
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)